src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -346,7 +436,59 @@ async function fetchAndCachePossibleCypressDomNode(
     }
   }
 
-  await cacheDomNodeEntry(possibleDomNodes, replayClient, pauseId, timeStampedPoint, testEvent);
+  let finalDomNodeDetails = {
+    possibleDomNodes,
+    pauseId,
+  };
+
+  // [PRO-599] If this is a click event, we need to find the DOM node as it existed
+  // _before_ the step
+  if (testEvent.data.command.name === "click") {
+    // If this is a click event, we need to find the DOM node as it
+    // existed _before_ the step, as it may have been disconnected
+    // during the step handling.
+    const beforeStep = testEvent.data.timeStampedPoints.beforeStep;
+
+    if (!beforeStep) {
+      return;
+    }
+
+    const persistentIds = possibleDomNodes.map(node => node.persistentId).filter(Boolean);
+
+    const { pauseId: beforeStepPauseId, domNodeIds } =
+      await domNodeIdForPersistentIdCache.readAsync(replayClient, beforeStep.point, persistentIds);
+
+    const parsedResultsWithValidNodeIds = domNodeIds.filter(({ domNodeId }) => !!domNodeId);
+
+    if (parsedResultsWithValidNodeIds.length === 0) {
+      return;
+    }
+
+    const matchingDomNodes = await Promise.all(
+      parsedResultsWithValidNodeIds.map(({ domNodeId }) =>
+        objectCache.readAsync(replayClient, beforeStepPauseId, domNodeId, "canOverflow")
+      )
+    );
+
+    // Note that in this case, we'll have the DOM nodes and pause ID from the before step,
+    // but we're going to put them in the cache with the result point as the key.
+    // This is okay! The test step row consistently uses the result point as the key,
+    // and this _is_ the DOM node and pause ID that relate to the step.
+    // `highlightNodes()` just needs these values, so it's fine if they're at a different
+    // point in time than the step itself.
```

Comment:
Does this mean we'll show the screenshot for the `result` point with a highlighter for the `beforeStep` point? We shouldn't be doing this, after all we're adding all this logic because the node may not be part of the DOM at the `result` point anymore, so we may be highlighting a node that doesn't exist in the screenshot. We should show the screenshot for the `beforeStep` point when hovering over click events instead.

Reply:
> Does this mean we'll show the screenshot for the `result` point with a highlighter for the `beforeStep` point?

I don't _think_ so. I think that execution point is just used as a _key_ to the cached data, but it doesn't actually _seek_ or update the timeline. It just tries to highlight DOM elements with the same id:

https://github.com/replayio/devtools/blob/96a68a9a1d104658d3078b65f18dd58709baefbc/src/ui/components/TestSuite/views/TestRecording/TestSectionRow.tsx#L148-L175

Which...come to think of it, since ids aren't globally unique outside of a pause, that seems like a potential problem.

Reply:
The details cache provides both the pause ID and the DOM node IDs, so there shouldn't be any collisions - we're targeting that specific pause for the highlighting.